packages/amplify_datastore/lib/amplify_datastore_stream_controller.dart

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'dart:async';
+import 'package:amplify_datastore/types/DataStoreHubEvents/DataStoreHubEvent.dart';
+import 'package:amplify_datastore/types/DataStoreHubEvents/ModelSyncedEvent.dart';
+import 'package:amplify_datastore/types/DataStoreHubEvents/NetworkStatusEvent.dart';
+import 'package:amplify_datastore/types/DataStoreHubEvents/OutboxMutationEvent.dart';
+import 'package:amplify_datastore/types/DataStoreHubEvents/SyncQueriesStartedEvent.dart';
+import 'package:amplify_core/types/hub/HubEventPayload.dart';
+import 'package:amplify_datastore_plugin_interface/amplify_datastore_plugin_interface.dart';
+
+import 'package:flutter/services.dart';
+
+import 'types/DataStoreHubEvents/OutboxStatusEvent.dart';
+
+EventChannel channel = const EventChannel("com.amazonaws.amplify/datastore_hub_events");
+ModelProviderInterface modelProvider;
+StreamSubscription eventStream;
+
+class DataStoreStreamController {
+  
+  StreamController get datastoreStreamController {
+    return controller;
+  }
+
+  void registerModelsForHub(ModelProviderInterface models) {
+    modelProvider = models;
+  }
+}
+
+StreamController controller = StreamController<DataStoreHubEvent>.broadcast(
+  onListen: onListen,
+  onCancel: onCancel,
+);
+
+onListen() {
```

Comment:
same as auth. Should we wrap these in the `DataStoreStreamController` class?

Reply:
I'm making _onListen private to tighten this up a bit, but trying to actually wrap all of this in a class yields this complaint:
```
The instance member '_authStreamController' can't be accessed in an initializer.
Try replacing the reference to the instance member with a different expression
```